In [1]:
import pandas as pd

In [2]:
LCS = pd.read_csv('projectData/LCS3_RESULT.csv', index_col=0, encoding='CP949')
LCS['version'] = LCS['version'].str.replace('버전','')
LCS['version'] = LCS['version'].str.replace(' ','')
LCS['version'] = LCS['version'].str.replace('<!----><!---->','')
ind = LCS[LCS['version'] == '11.5'].index
LCS = LCS.drop(index=ind)
LCS.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 905 entries, 0 to 669
Data columns (total 29 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   game_date          905 non-null    object
 1   version            905 non-null    object
 2   team1_name         905 non-null    object
 3   team1_score        905 non-null    int64 
 4   team2_score        905 non-null    int64 
 5   team2_name         905 non-null    object
 6   gametime           905 non-null    object
 7   blue_team          905 non-null    object
 8   blue_score         905 non-null    int64 
 9   blue_ban           905 non-null    object
 10  blue_object        905 non-null    object
 11  blue_object_count  905 non-null    int64 
 12  blue_call_name     905 non-null    object
 13  blue_line          905 non-null    object
 14  blue_champion      905 non-null    object
 15  blue_kill          905 non-null    int64 
 16  blue_death         905 non-null    object
 1

# LCS 리그별 오브젝트 확인

In [3]:
LCS_object = LCS[['blue_object','blue_object_count','red_object','red_object_count']]
LCS_object

blue_object  blue_object_count      red_object  red_object_count
0        turret@blue                  8      turret@red                 1
1     inhibitor@blue                  1   inhibitor@red                 0
2         baron@blue                  1       baron@red                 0
3        dragon@blue                  4      dragon@red                 0
4    riftHerald@blue                  2  riftHerald@red                 0
..               ...                ...             ...               ...
665      turret@blue                  4      turret@red                11
666   inhibitor@blue                  0   inhibitor@red                 3
667       baron@blue                  0       baron@red                 3
668      dragon@blue                  1      dragon@red                 6
669  riftHerald@blue                  1  riftHerald@red                 1

[905 rows x 4 columns]

In [4]:
a = LCS_object[LCS_object['blue_object'] == 'turret@blue']
aa = LCS_object[LCS_object['blue_object'] == 'inhibitor@blue']

In [5]:
LCS_turret = a.drop(columns=['blue_object','red_object'])
LCS_inhibitor = aa.drop(columns=['blue_object','red_object'])

In [6]:
LCS_turret.sum()

blue_object_count    1104
red_object_count     1107
dtype: int64

In [1]:
135+47

182

<PRE>
LCS는 블루팀일 때 1104개, 레드팀일때 1107개로
 LCS 서머 경기가 135경기 + 챔피언쉽 47경기

둘의 포탑 밀어내기 능력은 큰 차이가 없음    
블루팀은 경기당 6.0개
레드팀은 경기당 6.0개이며
    첫 포탑 획득률도 49: 51 로 서로 매우 비슷한 모습을 보임
    
포탑은 억제기포탑을 제외하면 거의 모든 포탑을 밀어내는 정도의 갯수임
 </PRE>

In [19]:
b = LCS_object[LCS_object['blue_object'] == 'riftHerald@blue']

In [20]:
LCS_riftHerald = b.drop(columns=['blue_object','red_object'])

In [21]:
LCS_riftHerald.sum()

blue_object_count    189
red_object_count     165
dtype: int64

블루나 레드의 전령 획득률이 크게 차이 나지 않는 것을 알 수 있고

블루가 전령을 획득했을 때 이를 바탕으로 포탑을 밀어내는 능력이 부족했던건 아닐까 생각함

블루팀이 먹는 전령은
경기당 1.0마리
레드팀의 경우 0.9마리의 전령을 처치함

전령은 게임에 두번 나옴

LCS는 전령을 서로 나눠먹게 되는 경우가 가장많다

이는 첫전령을 활용한 스노우볼 능력이 떨어지기에 발생하는 문제아닐까

In [11]:
c = LCS_object[LCS_object['blue_object'] == 'dragon@blue']
LCS_dragon= c.drop(columns=['blue_object','red_object'])

In [12]:
LCS_dragon.sum()

blue_object_count    372
red_object_count     461
dtype: int64

LCS의 경우도 LCK와 비슷하게 레드팀이 드래곤에서 우위를 가져감

경기당 블루팀은 2.0마리
레드팀은 2.5 마리

LCK에 비해 부족한 수치

In [24]:
d = LCS_object[LCS_object['blue_object'] == 'baron@blue']
LCS_baron= d.drop(columns=['blue_object','red_object'])

In [25]:
LCS_baron.sum()

blue_object_count    107
red_object_count     140
dtype: int64

 근데 또 바론의 경우 red가 크게 우위를 가지는 것을 알 수 있음<BR>
 LCK와 달리 바론은 왜 레드가 잘 먹는걸까
    </BR>
    
    블루팀의 경우 바론 획득은 경기당 0.58마리
    레드팀의 경우 0.77마리를 획득함
    
    LCS는 경기당 1.35마리
    

In [15]:
LCS['blue_champion'].value_counts().head(15)

리 신      54
비에고      49
이즈리얼     35
바루스      32
레넥톤      31
신 짜오     30
레오나      29
그웬       24
노틸러스     24
쓰레쉬      23
나르       21
다이애나     21
트리스타나    21
라칸       20
럼블       20
Name: blue_champion, dtype: int64

In [16]:
LCS['red_champion'].value_counts().head(15)

리 신      45
레오나      45
다이애나     40
비에고      37
바루스      35
노틸러스     32
트리스타나    28
칼리스타     27
아펠리오스    25
루시안      25
나르       22
신 짜오     22
레넥톤      21
브라움      20
녹턴       19
Name: red_champion, dtype: int64

In [17]:
LCS['blue_ban'].value_counts().head(5)

쓰레쉬     65
레넥톤     43
루시안     39
칼리스타    38
르블랑     35
Name: blue_ban, dtype: int64

드래곤의 부활시간은 5분이며 첫 드래곤 등장시간은 5분

LCS 평균 게임시간이 32분50초니까
첫 드래곤이후 약 5마리의 드래곤이 나오게 됨

첫 드래곤을 레드팀이 가져가는 경우가 63%


<pre>
지금까지 나온걸 정리해보면
LCS는 우선 경기당 12개의 포탑을 밀어내는 철거전을 하는데
첫 포탑 획득률도 서로 비슷한걸로 보아

1. 전령을 활용한 이후 포탑 철거에 능력이 부족하다
2. 전령을 통해 포탑을 철거하기 보다는 힘든 라인에 포탑골드를 획득한다

두가지의 경우를 생각할 수있음

또 특이점으로 전령을 경기당 블루,레드 모두 1마리를 가져오는 수치를 가짐
첫 전령을 블루가 획득하면 다음 전령을 레드가 획득하게 되는 경우가 많은거같고
이는 결국 전령을 이용한 포탑철거 능력이 비슷해지기에 포탑 갯수가 비슷하다고 생각됨
오히려 바론을 더 많이 먹는 레드팀이 결국 포탑 철거를 따라가거나 더 많이 하는거 아닐까


## 그렇다면 바론이 레드팀에서 더 많이 획득되는건 왜 그런가

블루팀이 가장 먼저 게임에 많은 영향을 줄 수 있는 첫 전령을 먹는 수치가 더 높지만
첫 타워 획득률이 블루,레드가 비슷하다는 점 등을 들어
 LCS의 팀들이 스노우볼 능력이 부족하다고 추측함

바론의 등장시간은 20:00
드래곤 첫 젠이후 꾸준한 처치가 이뤄졌다면
한팀이 독식한 경우 드래곤의 영혼을 바라볼 수도 있는 시간대임

레드팀의 첫 드래곤 획득률은 63%로 드래곤 스택에 대한 이점을 가진 레드팀이
바론 VS 드래곤 의 선택 상황에서 이점을 가지는 것이 아닐까 생각함

실제로 바론을 한마리만 먹은경우 레드나 블루나 둘다 비슷한 수치를 보여주지만 횟수는 레드가 많았고
두번째 바론을 레드가 모두 가져간 경우가 블루가 모두 가져간 경우에 비해 2배 많음

</pre>

LCS랑 경기할때는 전령을 주더라도 드래곤 스택만 잘신경쓰다면
크게 위험을 느낄 요소가없지않을까 

레드팀의 승률이 높음

운영적인 부분에서 실수만 안하면 LCK가 확굴려갈수있다

LCK가 전령을 획득한다면 본인들이 잘하는거 하면된다

In [36]:
LCS['red_ban'].value_counts().head(15)

신 짜오    87
레넥톤     47
쓰레쉬     47
칼리스타    42
그웬      40
비에고     39
르블랑     32
레오나     28
녹턴      26
노틸러스    26
리 신     24
브라움     24
바루스     23
이렐리아    22
아칼리     22
Name: red_ban, dtype: int64

In [37]:
LCS['blue_ban'].value_counts().head(15)

쓰레쉬     65
레넥톤     43
루시안     39
칼리스타    38
르블랑     35
그웬      35
바루스     34
비에고     30
리 신     29
아칼리     26
이렐리아    26
제이스     26
카르마     26
녹턴      25
다이애나    23
Name: blue_ban, dtype: int64

In [64]:
LCS.groupby(['blue_line','blue_champion'])[['game_date']].count().sort_values(by='game_date',ascending=False).head(10)

game_date
blue_line blue_champion           
AD        이즈리얼                  34
          바루스                   32
JG        신 짜오                  30
SUP       레오나                   29
TOP       레넥톤                   28
JG        비에고                   26
SUP       노틸러스                  24
TOP       그웬                    23
JG        리 신                   23
SUP       쓰레쉬                   23

In [ ]:
LCS

In [63]:
LCS.groupby(['blue_line','red_champion'])[['game_date']].count().sort_values(by='game_date',ascending=False).head(10)

game_date
blue_line red_champion           
SUP       레오나                  45
JG        다이애나                 40
AD        바루스                  35
SUP       노틸러스                 32
AD        아펠리오스                25
JG        리 신                  25
AD        칼리스타                 25
MID       루시안                  23
JG        신 짜오                 22
AD        트리스타나                22